### ISSUE TITLE
Updating to 1.10.0 failed

### ISSUE THREAD

**Issue by FrogoYT** [OctoPrint/OctoPrint#4998]
### The problem

I tried updating to 1.10.0 in safe mode but I the error message say:

Updating, please wait.

++++++++++++++++++++++++++++++++
Now updating OctoPrint to 1.10.0
++++++++++++++++++++++++++++++++
/usr/bin/python3 -m pip --disable-pip-version-check install https://github.com/OctoPrint/OctoPrint/archive/1.10.0.zip --no-cache-dir
Collecting https://github.com/OctoPrint/OctoPrint/archive/1.10.0.zip
Downloading https://github.com/OctoPrint/OctoPrint/archive/1.10.0.zip (6.4 MB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.4/6.4 MB 15.1 MB/s eta 0:00:00
Installing build dependencies: started
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: OctoPrint-FirmwareCheck>=2021.10.11 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2021.10.11)
Requirement already satisfied: semantic-version<3,>=2.10.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.10.0)
Collecting blinker<1.7.0,>=1.6.3
Downloading blinker-1.6.3-py3-none-any.whl (13 kB)
Requirement already satisfied: unidecode in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.3.8)
Requirement already satisfied: OctoPrint-FileCheck>=2021.2.23 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2024.3.27)
Requirement already satisfied: flask<2.3,>=2.2.3 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.2.5)
Requirement already satisfied: pkginfo<2,>=1.9.6 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.10.0)
Requirement already satisfied: feedparser<7,>=6.0.11 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (6.0.11)
Requirement already satisfied: cachelib<0.11,>=0.10.2 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.10.2)
Requirement already satisfied: passlib<2,>=1.7.4 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.7.4)
Requirement already satisfied: Flask-Limiter<4,>=3.5.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (3.5.1)
Requirement already satisfied: pydantic==1.10.12 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.10.12)
Requirement already satisfied: regex in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2022.10.31)
Collecting psutil<6,>=5.9.8
Downloading psutil-5.9.8.tar.gz (503 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 503.2/503.2 kB 5.6 MB/s eta 0:00:00
Installing build dependencies: started
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: requests<3,>=2.31.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.31.0)
Requirement already satisfied: colorlog<7,>=6.7.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (6.8.2)
Collecting websocket-client==1.6.1
Downloading websocket_client-1.6.1-py3-none-any.whl (56 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.9/56.9 kB 37.6 MB/s eta 0:00:00
Requirement already satisfied: wrapt<1.16,>=1.15 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.15.0)
Requirement already satisfied: frozendict<3,>=2.4.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.4.1)
Requirement already satisfied: Flask-Login<0.7,>=0.6.3 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.6.3)
Collecting markdown<3.5,>=3.4.4
Downloading Markdown-3.4.4-py3-none-any.whl (94 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 94.2/94.2 kB 39.5 MB/s eta 0:00:00
Requirement already satisfied: pytz in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2024.1)
Requirement already satisfied: Babel<2.13,>=2.12.1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.12.1)
Requirement already satisfied: pathvalidate<3,>=2.5.2 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.5.2)
Requirement already satisfied: sarge==0.1.7.post1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.1.7.post1)
Requirement already satisfied: class-doc<0.3,>=0.2.6 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.2.6)
Requirement already satisfied: future<1,>=0.18.3 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.18.3)
Requirement already satisfied: netaddr<0.9,>=0.8 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.8.0)
Requirement already satisfied: OctoPrint-PiSupport>=2023.10.10 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2023.10.10)
Requirement already satisfied: wheel in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (0.43.0)
Requirement already satisfied: setuptools in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (65.6.0)
Requirement already satisfied: emoji<3,>=2.10.1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.11.0)
Requirement already satisfied: Flask-Babel<4,>=3.1.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (3.1.0)
Collecting netifaces2<0.1,>=0.0.21
Downloading netifaces2-0.0.22-cp37-abi3-musllinux_1_1_aarch64.whl (1.2 MB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 11.5 MB/s eta 0:00:00
Requirement already satisfied: filetype<2,>=1.2.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.2.0)
Requirement already satisfied: pylru<2,>=1.2.1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.2.1)
Collecting tornado<6.3,>=6.2
Downloading tornado-6.2-cp37-abi3-musllinux_1_1_aarch64.whl (426 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 426.5/426.5 kB 13.3 MB/s eta 0:00:00
Requirement already satisfied: werkzeug<2.3,>=2.2.3 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.2.3)
Collecting zeroconf~=0.127
Downloading zeroconf-0.132.2.tar.gz (170 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 170.7/170.7 kB 42.6 MB/s eta 0:00:00
Installing build dependencies: started
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: pyserial<4,>=3.5 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (3.5)
Requirement already satisfied: Flask-Assets<3,>=2.1.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.1.0)
Collecting argon2-cffi>=23.1.0
Downloading argon2_cffi-23.1.0-py3-none-any.whl (15 kB)
Requirement already satisfied: zipstream-ng<2.0.0,>=1.7.1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.7.1)
Requirement already satisfied: sentry-sdk<2,>=1.40.0 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (1.45.0)
Requirement already satisfied: Click<9,>=8.1.7 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (8.1.7)
Requirement already satisfied: PyYAML<7,>=6.0.1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (6.0.1)
Requirement already satisfied: watchdog<3,>=2.3.1 in /usr/lib/python3.10/site-packages (from OctoPrint==1.10.0) (2.3.1)
Requirement already satisfied: typing-extensions>=4.2.0 in /usr/lib/python3.10/site-packages (from pydantic==1.10.12->OctoPrint==1.10.0) (4.11.0)
Collecting argon2-cffi-bindings
Downloading argon2_cffi_bindings-21.2.0-cp36-abi3-musllinux_1_1_aarch64.whl (83 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.6/83.6 kB 42.6 MB/s eta 0:00:00
Requirement already satisfied: more-itertools>=5.0.0 in /usr/lib/python3.10/site-packages (from class-doc<0.3,>=0.2.6->OctoPrint==1.10.0) (10.2.0)
Requirement already satisfied: sgmllib3k in /usr/lib/python3.10/site-packages (from feedparser<7,>=6.0.11->OctoPrint==1.10.0) (1.0.0)
Requirement already satisfied: itsdangerous>=2.0 in /usr/lib/python3.10/site-packages (from flask<2.3,>=2.2.3->OctoPrint==1.10.0) (2.1.2)
Requirement already satisfied: Jinja2>=3.0 in /usr/lib/python3.10/site-packages (from flask<2.3,>=2.2.3->OctoPrint==1.10.0) (3.1.3)
Requirement already satisfied: webassets>=2.0 in /usr/lib/python3.10/site-packages (from Flask-Assets<3,>=2.1.0->OctoPrint==1.10.0) (2.0)
Requirement already satisfied: limits>=2.8 in /usr/lib/python3.10/site-packages (from Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (3.10.1)
Requirement already satisfied: rich<14,>=12 in /usr/lib/python3.10/site-packages (from Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (13.7.1)
Requirement already satisfied: ordered-set<5,>4 in /usr/lib/python3.10/site-packages (from Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (4.1.0)
Requirement already satisfied: charset-normalizer<4,>=2 in /usr/lib/python3.10/site-packages (from requests<3,>=2.31.0->OctoPrint==1.10.0) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3.10/site-packages (from requests<3,>=2.31.0->OctoPrint==1.10.0) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/lib/python3.10/site-packages (from requests<3,>=2.31.0->OctoPrint==1.10.0) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python3.10/site-packages (from requests<3,>=2.31.0->OctoPrint==1.10.0) (2024.2.2)
Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/lib/python3.10/site-packages (from werkzeug<2.3,>=2.2.3->OctoPrint==1.10.0) (2.1.5)
Requirement already satisfied: ifaddr>=0.1.7 in /usr/lib/python3.10/site-packages (from zeroconf~=0.127->OctoPrint==1.10.0) (0.2.0)
Requirement already satisfied: async-timeout>=3.0.0 in /usr/lib/python3.10/site-packages (from zeroconf~=0.127->OctoPrint==1.10.0) (4.0.3)
Requirement already satisfied: packaging<25,>=21 in /usr/lib/python3.10/site-packages (from limits>=2.8->Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (24.0)
Requirement already satisfied: deprecated>=1.2 in /usr/lib/python3.10/site-packages (from limits>=2.8->Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (1.2.14)
Requirement already satisfied: importlib-resources>=1.3 in /usr/lib/python3.10/site-packages (from limits>=2.8->Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (6.4.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/lib/python3.10/site-packages (from rich<14,>=12->Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (2.17.2)
Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/lib/python3.10/site-packages (from rich<14,>=12->Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (3.0.0)
Collecting cffi>=1.0.1
Downloading cffi-1.16.0.tar.gz (512 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 512.9/512.9 kB 12.1 MB/s eta 0:00:00
Installing build dependencies: started
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Installing backend dependencies: started
Installing backend dependencies: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'done'
Collecting pycparser
Downloading pycparser-2.22-py3-none-any.whl (117 kB)
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 117.6/117.6 kB 26.9 MB/s eta 0:00:00
Requirement already satisfied: mdurl~=0.1 in /usr/lib/python3.10/site-packages (from markdown-it-py>=2.2.0->rich<14,>=12->Flask-Limiter<4,>=3.5.0->OctoPrint==1.10.0) (0.1.2)
Building wheels for collected packages: OctoPrint, psutil, zeroconf, cffi
Building wheel for OctoPrint (pyproject.toml): started
Building wheel for OctoPrint (pyproject.toml): finished with status 'done'
Created wheel for OctoPrint: filename=OctoPrint-1.10.0-py2.py3-none-any.whl size=3361572 sha256=17821125f51a7b0063ff6a038a676fb9f067f43f318093af781f6cb0101e86b0
Stored in directory: /tmp/pip-ephem-wheel-cache-3762frdx/wheels/fe/cb/bb/4f7623338e7016c80acc82ffcfac7b0102745461e69453058e
Building wheel for psutil (pyproject.toml): started
Building wheel for psutil (pyproject.toml): finished with status 'error'
Building wheel for zeroconf (pyproject.toml): started
error: subprocess-exited-with-error

× Building wheel for psutil (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> [46 lines of output]
running bdist_wheel
running build
running build_py
creating build
creating build/lib.linux-aarch64-cpython-310
creating build/lib.linux-aarch64-cpython-310/psutil
copying psutil/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_common.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_compat.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_psaix.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_psbsd.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_pslinux.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_psosx.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_psposix.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_pssunos.py -> build/lib.linux-aarch64-cpython-310/psutil
copying psutil/_pswindows.py -> build/lib.linux-aarch64-cpython-310/psutil
creating build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/__init__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/__main__.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/runner.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_aix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_bsd.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_connections.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_contracts.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_linux.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_memleaks.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_misc.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_osx.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_posix.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_process.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_process_all.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_sunos.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_system.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_testutils.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_unicode.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
copying psutil/tests/test_windows.py -> build/lib.linux-aarch64-cpython-310/psutil/tests
running build_ext
building 'psutil._psutil_linux' extension
creating build/temp.linux-aarch64-cpython-310
creating build/temp.linux-aarch64-cpython-310/psutil
creating build/temp.linux-aarch64-cpython-310/psutil/arch
creating build/temp.linux-aarch64-cpython-310/psutil/arch/linux
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Os -fomit-frame-pointer -g -O2 -Os -fomit-frame-pointer -g -O2 -Os -fomit-frame-pointer -g -O2 -DTHREAD_STACK_SIZE=0x100000 -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=598 -DPy_LIMITED_API=0x03060000 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -DPSUTIL_LINUX=1 -I/usr/include/python3.10 -c psutil/_psutil_common.c -o build/temp.linux-aarch64-cpython-310/psutil/_psutil_common.o
psutil could not be installed from sources because gcc is not installed. Try running:
sudo apk add gcc python3-dev musl-dev linux-headers
error: command 'gcc' failed: No such file or directory
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed building wheel for psutil
Building wheel for zeroconf (pyproject.toml): still running...
Building wheel for zeroconf (pyproject.toml): still running...
Building wheel for zeroconf (pyproject.toml): finished with status 'done'
Created wheel for zeroconf: filename=zeroconf-0.132.2-cp310-cp310-musllinux_1_2_aarch64.whl size=115794 sha256=7a5a0362a7c4b238a04c7ccb6b0fc8fbdb1e3536e6d32c0b8fae622624f39eac
Stored in directory: /tmp/pip-ephem-wheel-cache-3762frdx/wheels/98/47/82/fc68c1a1ff18ee6a65dc6862a039bd3342c89b2807d9ac402f
Building wheel for cffi (pyproject.toml): started
error: subprocess-exited-with-error

× Building wheel for cffi (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> [48 lines of output]

No working compiler found, or bogus compiler options passed to
the compiler from Python's standard "distutils" module.  See
the error messages above.  Likely, the problem is not related
to CFFI but generic to the setup.py of any Python package that
tries to compile C code.  (Hints: on OS/X 10.8, for errors about
-mno-fused-madd see http://stackoverflow.com/questions/22313407/
Otherwise, see https://wiki.python.org/moin/CompLangPython or
the IRC channel #python on irc.libera.chat.)

Trying to continue anyway.  If you are trying to install CFFI from
a build done in a different context, you can ignore this warning.

running bdist_wheel
running build
running build_py
creating build
creating build/lib.linux-aarch64-cpython-310
creating build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/__init__.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/_imp_emulation.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/_shimmed_dist_utils.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/api.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/backend_ctypes.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/cffi_opcode.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/commontypes.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/cparser.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/error.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/ffiplatform.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/lock.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/model.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/pkgconfig.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/recompiler.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/setuptools_ext.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/vengine_cpy.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/vengine_gen.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/verifier.py -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/_cffi_include.h -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/parse_c_type.h -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/_embedding.h -> build/lib.linux-aarch64-cpython-310/cffi
copying src/cffi/_cffi_errors.h -> build/lib.linux-aarch64-cpython-310/cffi
running build_ext
building '_cffi_backend' extension
creating build/temp.linux-aarch64-cpython-310
creating build/temp.linux-aarch64-cpython-310/src
creating build/temp.linux-aarch64-cpython-310/src/c
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Os -fomit-frame-pointer -g -O2 -Os -fomit-frame-pointer -g -O2 -Os -fomit-frame-pointer -g -O2 -DTHREAD_STACK_SIZE=0x100000 -fPIC -DFFI_BUILDING=1 -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python3.10 -c src/c/_cffi_backend.c -o build/temp.linux-aarch64-cpython-310/src/c/_cffi_backend.o
error: command 'gcc' failed: No such file or directory
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
ERROR: Failed building wheel for cffi
ERROR: Could not build wheels for psutil, cffi, which is required to install pyproject.toml-based projects
Building wheel for cffi (pyproject.toml): finished with status 'error'
Successfully built OctoPrint zeroconf
Failed to build psutil cffi

The update did not finish successfully. Please consult <code>octoprint.log</code> and <code>plugin_softwareupdate_console.log</code> for details.
![image](https://github.com/OctoPrint/OctoPrint/assets/168278532/dcbd1192-d0d7-4d84-b732-97de5a3ee0ef)

### Did the issue persist even in safe mode?

Yes, it did persist

### Version of OctoPrint

1.9.3

### Operating system running OctoPrint

OctoPrint 1.9.3 Python 3.10.14

### Printer model & used firmware incl. version

Creality Ender 3 V3 SE, 1.0.6

### Browser and version of browser, operating system running browser

Google Chrome, Windows

### Checklist of files to include below

- [X] Systeminfo Bundle (always include!)
- [ ] Contents of the JavaScript browser console (always include in cases of issues with the user interface)
- [ ] Screenshots and/or videos showing the problem (always include in case of issues with the user interface)
- [ ] GCODE file with which to reproduce (always include in case of issues with GCODE analysis or printing behaviour)

### Additional information & file uploads

[octoprint-systeminfo-20240427123730.zip](https://github.com/OctoPrint/OctoPrint/files/15138183/octoprint-systeminfo-20240427123730.zip)

**Comment by cp2004** [comment]
> Operating system running OctoPrint
OctoPrint 1.9.3 Python 3.10.14

Neither of these things are an operating system.

Your OS looks to be Linux, but OctoPrint is installed globally rather than in a virtual environment, so this is not a typical setup. Please provide more details about your setup.

**Comment by FrogoYT** [comment]
Oh, I am new to octoprint and GitHub, but yeah, you're right. I'm using an android phone to power octoprint. Sorry I forgot to include that.

**Comment by foosel** [comment]
I'm sorry to say that this is an issue with your chosen environment. You are missing a compiler for some native parts of third party dependencies. 

This is not an issue with OctoPrint but an issue with the environment you are trying to set it up under.

You might want to look at octo4a.

**Comment by cp2004** [comment]
https://github.com/feelfreelinux/octo4a/releases/tag/1.2.6

Looks like this should have been fixed with Octo4a 1.2.6, released an hour ago.

**Comment by zaventh** [comment]
For posterity... I had this issue on Octo4a. It appears the "fix" there is to recreate the entire chroot and reinstall OctoPrint which I wished to avoid. The main issue is 1.10.0 requires `psutil >=5.9.8 <6` while the Octo4a chroot Alpine 3.17 can only provide psutil 5.9.4.

I was able to resolve it by shelling into the chroot and adding build tools and other new dependencies at the dist level:

```shell
apk add build-base gcc python3-dev musl-dev linux-headers py3-zeroconf py3-cffi
apk del ps3-psutil` # removes dist version of psutil at 5.9.4
```
At which point re-running the installer from the OctoPrint UI builds the psutil wheel correctly in the environment.

Simply updating the Octo4a app did not solve the issue for me on an existing installation.
